omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/converter/openmrs2_2/CodedOrFreeTextConverter.java

```diff
@@ -30,25 +32,37 @@ public class CodedOrFreeTextConverter extends BaseDelegatingConverter<CodedOrFre
 	
 	@Override
 	public SimpleObject asRepresentation(CodedOrFreeText instance, Representation rep) throws ConversionException {
-		SimpleObject codedOfFreeText = new SimpleObject();
-		
-		if (instance.getSpecificName() != null) {
-			ConceptNameResource1_9 conceptNameResource = (ConceptNameResource1_9) Context.getService(RestService.class)
-			        .getResourceBySupportedClass(ConceptName.class);
-			codedOfFreeText.add("specificName", conceptNameResource.asRepresentation(instance.getSpecificName(), rep));
-		}
-		
-		if (instance.getCoded() != null) {
-			ConceptResource2_0 conceptResource = (ConceptResource2_0) Context.getService(RestService.class)
-			        .getResourceBySupportedClass(Concept.class);
-			codedOfFreeText.add("coded", conceptResource.asRepresentation(instance.getCoded(), rep));
+		SimpleObject codedOrFreeText = new SimpleObject();
+		if (rep instanceof CustomRepresentation) {
```

Comment:
I don't know why the framework doesn't automatically handle this, but it doesn't seem to.  So here, I'm explicitly parsing the custom representation and using that to determine which properties of the CodedOrNonCoded to convert, and how.